op-e2e/actions/user.go

```diff
@@ -149,6 +149,13 @@ func (s *BasicUser[B]) ActRandomTxToAddr(t Testing) {
 	s.txToAddr = to
 }
 
+func (s *BasicUser[B]) ActSetTxCalldata(calldata []byte) Action {
+	return func(t Testing) {
+		require.NotNil(t, calldata)
+		s.txCallData = calldata
```

Comment:
two things I just found by reading this code. I don't think we actually send `s.txCalldata` (line 205/212 - it seems to only be used in the gas estimation). Second is that if we require non-nill calldata we should have a way to reset the calldata b/c a basic user can send multiple transactions.

Reply:
Upon further review, we set use txCalldata with deposits for a crossLayer user but not for a basic user & still don't have a way to zero it (other than length 0 but non nil data array).

Reply:
Good catch - I was going crazy because things were not going as expected when writing the tests and couldn't figure out why